src/utilities/_hypre_utilities.h

```diff
@@ -1948,18 +1948,23 @@ HYPRE_Int hypre_IntArrayCopy( hypre_IntArray *x, hypre_IntArray *y );
 hypre_IntArray* hypre_IntArrayCloneDeep_v2( hypre_IntArray *x,
                                             HYPRE_MemoryLocation memory_location );
 hypre_IntArray* hypre_IntArrayCloneDeep( hypre_IntArray *x );
+HYPRE_Int hypre_IntArrayMigrate( hypre_IntArray *v, HYPRE_MemoryLocation memory_location );
+HYPRE_Int hypre_IntArrayPrint( MPI_Comm comm, hypre_IntArray *array, const char *filename );
+HYPRE_Int hypre_IntArrayRead( MPI_Comm comm, const char *filename, hypre_IntArray **array_ptr );
 HYPRE_Int hypre_IntArraySetConstantValuesHost( hypre_IntArray *v, HYPRE_Int value );
 HYPRE_Int hypre_IntArraySetConstantValues( hypre_IntArray *v, HYPRE_Int value );
 HYPRE_Int hypre_IntArrayCountHost( hypre_IntArray *v, HYPRE_Int value,
                                    HYPRE_Int *num_values_ptr );
 HYPRE_Int hypre_IntArrayCount( hypre_IntArray *v, HYPRE_Int value,
                                HYPRE_Int *num_values_ptr );
+HYPRE_Int hypre_IntArrayInverseMapping( hypre_IntArray *v, hypre_IntArray **w_ptr );
```

Comment:
Should this be part of the `IntArray` "class"?  It only makes sense if the original IntArray defines a permutation (a 1-1 mapping of the set of indices).  So maybe this should be renamed?

Reply:
I see the permutation as a special case of an `IntArray`, so I think the naming makes sense. We could perhaps add a new flag member to `IntArray` that indicates it's a permutation, something like `is_permutation`. If that flag is false, then `IntArrayInverseMapping` is a no-op. What is your opinion?

Reply:
Another idea is to rename it `hypre_IntArrayPermInverseMapping` to indicate it operates on a special type of `IntArray`